File: BACKEND/src/models/PswRecord.rs

use serde::{Deserialize, Serialize};



#[derive(Debug, Deserialize, Serialize, Clone)]
pub struct PswRecord {
  pub owner_id: String,

  pub record_id: String,

  pub app_ico: Option<String>,
  pub app_name: String,

  pub account_name: String,
  pub encoded_login: String,
  pub encoded_password: String,

  pub tags: Vec<String>,

  pub created_at: String,
}




#[derive(Debug, Deserialize, Serialize, Clone)]
pub struct NewPswRecordDTO {
  //pub owner_id: String,

  //pub record_id: String,

  pub app_ico: Option<String>,
  pub app_name: String,

  pub account_name: String,
  pub encoded_login: String,
  pub encoded_password: String,

  pub tags: Vec<String>,

  //pub created_at: String,
}

File: FRONTEND/src/types/PswRecord.ts


export interface TypedPswRecord {

  owner_id: string,
  record_id: string,
  app_ico: string | null,
  app_name: string,
  account_name: string,
  encoded_login: string,
  encoded_password: string,
  tags: string[],
  created_at: string,

  _record_type: "TYPED",
}




export interface RawPswRecord {

  owner_id: string,
  record_id: string,
  app_ico: string | null,
  app_name: string,
  raw_content: string,
  tags: string[],
  created_at: string,

  _record_type: "RAW",
}